source/NVDAObjects/IAccessible/ia2Web.py

```diff
@@ -46,6 +46,21 @@ def _get_shouldCreateTreeInterceptor(self):
 class Application(Document):
 	shouldCreateTreeInterceptor = False
 
+class WebDialog(Document):
+	"""
+	A dialog that will use a treeInterceptor if its parent currently does.
+	"""
+
+	def _get_shouldCreateTreeInterceptor(self):
+		if self.parent.treeInterceptor:
+			return True
+		return False
+
+	# For dialogs that will get browseMode and are focusable themselves (have a tabindex):
+	# Allow setting focus to the dialog directly, rather than its first focusable descendant
+	def _get__alwaysFocusFirstfocusableDescendant(self):
+		return not self.shouldCreateTreeInterceptor

```

Comment:
As discussed, unless I'm missing something, I don't think the changes related to FocusableUnfocusableContainer are necessary. That behavior should only get used if the container is not focusable itself.